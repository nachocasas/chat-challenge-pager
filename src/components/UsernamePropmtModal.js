import React, { Component } from 'react';

export default class UsernamePropmtModal extends Component {

  constructor(props){
    super(props);
    this.state = {
      text: ''
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleEnter = this.handleEnter.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({
      text: e.target.value
    })
  }

  handleEnter(e){
    const code = (e.keyCode ? e.keyCode : e.which)
    if(code == 13){
      this.handleSubmit();
    }
  }

  handleSubmit(){
    this.props.onUsernameEnter(this.state.text);
    this.setState({ text: '' });
  }

  render(){
    return(
      <div className="username-modal">
        <div className='prompt'>
          <h3>Please enter your username:</h3>
          <input onChange={ this.handleChange } onKeyPress={ this.handleEnter }></input>
          <button onClick={ this.handleSubmit }>Chat!</button>
        </div>
      </div>
    )
  }
}