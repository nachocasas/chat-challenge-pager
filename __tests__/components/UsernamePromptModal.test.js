import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UsernamePropmtModal from '../../src/components/UsernamePropmtModal';

describe('<UsernamePropmtModal />', () => {
  describe('render()', () => {

    test('render component', () => {
      const wrapper = shallow(<UsernamePropmtModal />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });
});