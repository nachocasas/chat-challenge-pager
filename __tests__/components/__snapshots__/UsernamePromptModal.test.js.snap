// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsernamePropmtModal /> render() render component 1`] = `
<div
  className="username-modal"
>
  <div
    className="prompt"
  >
    <h3>
      Please enter your username:
    </h3>
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
    />
    <button
      onClick={[Function]}
    >
      Chat!
    </button>
  </div>
</div>
`;
